### ISSUE TITLE
There are no access keys in the macOS keychain

### ISSUE THREAD

**Issue by hanakannzashi** [near/near-cli-rs#160]
I do set keychain in my macos account, but keychain can not be found. Are there any operation tutorial?

**Comment by hanakannzashi** [comment]
Also, when I add key in my account with generated key pair, and save it to my macos keychain, but, nothing found in keychain actually.

**Comment by frol** [comment]
@hanakannzashi Have you seen "The data for the access key is saved in macOS Keychain" message when you added the key?

Here is a demo of importing an existing account using a seed phrase and storing it in macOS keychain. Then, I send NEAR tokens from that account just to show that I can sign the transaction with the key stored in macOS keychain.

<img width="1432" alt="Screenshot 2023-01-17 at 21 50 01" src="https://user-images.githubusercontent.com/304265/213010477-8edef290-2669-4fe4-a7a7-6114d2ef4bff.png">

You can see that the public key of the imported key matches the public key that was used to sign the transaction.

**Comment by hanakannzashi** [comment]
> @hanakannzashi Have you seen "The data for the access key is saved in macOS Keychain" message when you added the key?
> 
> Here is a demo of importing an existing account using a seed phrase and storing it in macOS keychain. Then, I send NEAR tokens from that account just to show that I can sign the transaction with the key stored in macOS keychain.
> 
> <img alt="Screenshot 2023-01-17 at 21 50 01" width="1432" src="https://user-images.githubusercontent.com/304265/213010477-8edef290-2669-4fe4-a7a7-6114d2ef4bff.png">
> 
> You can see that the public key of the imported key matches the public key that was used to sign the transaction.

I didn't see 'The data for the access key is saved in macOS Keychain' message.
<img width="1166" alt="p" src="https://user-images.githubusercontent.com/95295094/213074279-8bb47c6f-a9e0-43f0-a22e-6df4b3fc8485.png">

**Comment by hanakannzashi** [comment]
I found that key is not display on macos keychain UI but truly saved, so how can I view my all keys stored in keychain?

**Comment by frol** [comment]
> I add new key on my account cool8.testnet, using autogenerate-new-keypair save-to-macos-keychain, I did not see message "The data for the access key is saved in macOS Keychain"

Yeah, the message is different, but the key is actually stored in macOS keychain after all. It is worth aligning the messages, and review all the messages in the CLI in general (I know there are some typos, copy-paste errors, and unclear messages here and there, but I did not have time to sit down and review all of them).

> when I sign transaction with macos-keychain again, it still using the key I imported in step 1, new key pair seems lost.

CLI fetches the list of full access keys for a given account from RPC (public keys) and probes the keychain to find the corresponding private key, and once one is found, it is used. In your case, the older key was probed earlier and thus it was used. Try removing the first key from the keychain and you should see that it would use another key.

I cannot think of a use case where users would need to have several full access keys to the same account on their computer and they would care which one is actually used.

> How can I view all my keys saved in macos-keychain with near-cli-rs? or other simple way? Keychain Manager App is hard to use.

There is no such functionality yet. Probing macOS keychain is not efficient as it may ask user's approval every time we try to read from the keychain, so this functionality requires study and design phase.

> Can keys be shared on different mac devices through cloud?

I believe they are shared, but I don't have a way to test whether it is actually the case myself as I don't have another device with the same icloud. Can you?

**Comment by hanakannzashi** [comment]
I remember some keys doesn't share through iCloud, but I don't know the condition, also I don't have another device to test it. Thank you for your answer!
